castle/cms/services/google/analytics.py

```diff
@@ -9,20 +9,27 @@ def get_ga_service(ga_scope=['https://www.googleapis.com/auth/analytics.readonly
     api_email = registry.get('castle.google_api_email', None)
     api_key = registry.get('castle.google_api_service_key_file', None)
     ga_id = registry.get('castle.google_analytics_id', None)
+    ua_id = registry.get('castle.universal_analytics_id', None)
 
-    if not api_key or not api_email or not ga_id:
+    if not api_key or not api_email or (not ga_id and not ua_id):
         return
-
-    # Authenticate and construct service.
-    return get_service(
-        'analytics', 'v3', ga_scope, b64decode_file(api_key)[1], api_email)
+    if ga_id:
+        return get_service(
+            'analytics', 'v4', ga_scope, b64decode_file(api_key)[1], api_email)
+    elif ua_id:
```

Comment:
the castle.universal_analytics_id registry config item is what is backwards compatible with v3 analytics? the castle.universal_analytics_id is a new field, correct? where was that ID value held before? in the castle.google_analytics_id field? is part of the upgrade step migrating the GA3 compatible key in google_analytics_id field to the universal_analytics_id field?

if not, we should make sure to migrate that value so that after this update is applied, sites will still function correctly without having to go into each site configuration to re-configure the GA3 id